tests/bluetooth/host/conn/src/main.c

```diff
@@ -0,0 +1,134 @@
+
+/*
+ * Test that bt_conn_le_create() returns -EINVAL if conn is not NULL and
+ * CONFIG_BT_CONN_CHECK_NULL_BEFORE_CREATE is enabled.
+ *
+ * The test must be compiled with and without CONFIG_BT_CONN_CHECK_NULL_BEFORE_CREATE
+ * to ensure that the -EINVAL error is returned only when this Kconfig option is enabled.
+ */
+ZTEST(conn, test_bt_conn_le_create_check_null_conn)
+{
+	bt_addr_le_t peer = {.a.val = {0x01}};
+	const struct bt_conn_le_create_param create_param = {
+		.options = BT_CONN_LE_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+		.interval_coded = 0,
+		.window_coded = 0,
+		.timeout = 100UL * MSEC_PER_SEC / 10,
+	};
+	struct bt_conn *conn;
+	int err;
+
+	/* Set conn to any non-NULL value to trigger the check. */
+	conn = (struct bt_conn *)1;
+
+	err = bt_conn_le_create(&peer, &create_param, BT_LE_CONN_PARAM_DEFAULT, &conn);
+	/* err must not be -EINVAL if CONFIG_BT_CONN_CHECK_NULL_BEFORE_CREATE is not enabled,
+	 * otherwise -EAGAIN is returned.
+	 *
+	 * The printk is used to see what actually was compiled.
+	 */
+#if defined(CONFIG_BT_CONN_CHECK_NULL_BEFORE_CREATE)
+	printk("Expected error -EINVAL\n");
+	zassert_equal(err, -EINVAL, "Failed starting initiator (err %d)", err);
+#else
+	printk("Expected error -EAGAIN\n");
+	zassert_equal(err, -EAGAIN, "Failed starting initiator (err %d)", err);
+#endif
+
+	/* If the conn is NULL, next error must not be -EINVAL. */
+	conn = NULL;
+	err = bt_conn_le_create(&peer, &create_param, BT_LE_CONN_PARAM_DEFAULT, &conn);
+	zassert_equal(err, -EAGAIN, "Failed starting initiator (err %d)", err);
+}
+
+/*
+ * Test that bt_conn_le_create_synced() returns -EINVAL if conn is not NULL and
+ * CONFIG_BT_CONN_CHECK_NULL_BEFORE_CREATE is enabled.
+ *
+ * The test must be compiled with and without CONFIG_BT_CONN_CHECK_NULL_BEFORE_CREATE
+ * to ensure that the -EINVAL error is returned only when this Kconfig option is enabled.
+ */
+ZTEST(conn, test_bt_conn_le_create_synced_check_null_conn)
+{
+	bt_addr_le_t peer = {.a.val = {0x01}};
+	struct bt_le_conn_param conn_param = {
+		.interval_min = 0x30,
+		.interval_max = 0x30,
+		.latency = 0,
+		.timeout = 400,
+	};
+	struct bt_conn_le_create_synced_param synced_param = {
+		.peer = &peer,
+	};
+	struct bt_le_ext_adv *adv = NULL;
+	struct bt_conn *conn;
+	int err;
+
+	/* Set conn to any non-NULL value to trigger the check. */
+	conn = (struct bt_conn *)1;
+	err = bt_conn_le_create_synced(adv, &synced_param, &conn_param, &conn);
+
+	/* err must not be -EINVAL if CONFIG_BT_CONN_CHECK_NULL_BEFORE_CREATE is not enabled,
+	 * otherwise -EAGAIN is returned.
+	 *
+	 * The printk is used to see what actually was compiled.
+	 */
+#if defined(CONFIG_BT_CONN_CHECK_NULL_BEFORE_CREATE)
+	printk("Expected error -EINVAL\n");
+	zassert_equal(err, -EINVAL, "Failed starting initiator (err %d)", err);
+#else
+	printk("Expected error -EAGAIN\n");
+	zassert_equal(err, -EAGAIN, "Failed starting initiator (err %d)", err);
```

Comment:
Why is `EAGAIN` returned here?

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/6f937c9ceaf67397189fd2e1275ff8eba264c801/subsys/bluetooth/host/conn.c#L3622-L3628

Reply:
It is the first error code after these 2 new `CHECKIF`s.

Reply:
Ah. That is neither documented nor described in comments is it? Please add it to either the documentation of the function, or at the very least document the test here that this is testing an assumed return value (just in case that the return values of the function change and that people in the future can determine if to modify the test or what to do if it starts failing)